python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -718,6 +718,12 @@ def can_reconcile_candidate(candidate: AssetKeyPartitionKey) -> bool:
         instance=instance_queryer.instance,
     )
 
+    def will_be_materialized_for_freshness(asset_partition: AssetKeyPartitionKey) -> bool:
+        return asset_partition in conditions_by_asset_partition_for_freshness and all(
+            condition.decision_type == AutoMaterializeDecisionType.MATERIALIZE
```

Comment:
Maybe should included DISCARD? Let's say C is partitioned and hits the max partitions limit, D would still double fire right?

Reply:
technically, a partitioned asset will never have a ForFreshness reason, as only unpartitioned assets are handled by that logic, so that specific scenario isn't possible.

agree with the high level feedback that there's weird stuff that can happen at these edges